### PR TITLE
Add Event.getAsMap() function to return the Event as a Map<Object, Object>

### PR DESCRIPTION
**Pull request by graytaylor0**
Signed-off-by: Taylor Gray <33740195+graytaylor0@users.noreply.github.com>

### Description
* Adds a function `getAsMap()` to the Event interface
* Implements this function in JacksonEvent and returns the Event as a `Map<Object, Object>` using  ObjectMapper
 
### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-api/src/main/java/com/amazon/dataprepper/model/event/JacksonEvent.java [hunk 1]
```diff
@@ -272,6 +272,11 @@ public boolean isValueAList(final String key) {
         return node.isArray();
     }
 
+    @Override
+    public Map<Object, Object> getAsMap() {
+        return mapper.convertValue(jsonNode, new TypeReference<Map<Object, Object>>() {});
```
Comment (chenqi0805) [inline data-prepper-api/src/main/java/com/amazon/dataprepper/model/event/JacksonEvent.java:277]: May consider define the type reference instance as constant field instead of instantiating new instance at every call.

**Review by chenqi0805 — DISMISSED** [review]

File: data-prepper-api/src/main/java/com/amazon/dataprepper/model/event/Event.java [hunk 2]
```diff
@@ -85,4 +86,10 @@
      * @since 1.2
      */
     boolean isValueAList(String key);
+
+    /**
+     * @return a Map representation of the Event
+     * @since 1.3
+     */
+    Map<Object, Object> getAsMap();
```
Comment (dlvenable) [inline data-prepper-api/src/main/java/com/amazon/dataprepper/model/event/Event.java:94]: Our model guarantees that the key is a string. So, I think we can return `Map<String, Object>` here instead.

Comment (dlvenable) [inline data-prepper-api/src/main/java/com/amazon/dataprepper/model/event/Event.java:94]: I'd avoid using a `get` method here since it sounds like a getter. You could rename this to either `toMap()` or `asMap()`. I'd prefer `toMap()` because it is consistent with other Java APIs.

File: data-prepper-api/src/test/java/com/amazon/dataprepper/model/event/JacksonEventTest.java [hunk 3]
```diff
@@ -312,6 +313,26 @@ public void testToString_withSimpleObject() {
         assertThat(result, is(equalTo(String.format("{\"foo\":\"bar\",\"testObject\":{\"field1\":\"%s\"},\"list\":[1,4,5]}", value))));
     }
 
+    @Test
+    public void testGetAsMap_with_EmptyData() {
```
Comment (dlvenable) [inline data-prepper-api/src/test/java/com/amazon/dataprepper/model/event/JacksonEventTest.java:317]: Thanks for thinking of edge cases like this one!

**Review by dlvenable — CHANGES_REQUESTED** [review]

**Review by chenqi0805 — APPROVED** [review]

**Review by dlvenable — APPROVED** [review]
